fix for CDPD-6438

amitagarwal06 opened the pull request:
CDPD-6438: Have tuned yarn, mapreduce & tez configs in DE & DE HA template as per sub-task JIRA for https://jira.cloudera.com/browse/CDPD-6438

List of tuned configs are -
YARN
------
yarn.scheduler.capacity.root.default.minimum-user-limit-percent=100
yarn.scheduler.capacity.maximum-am-resource-percent=0.33
yarn.scheduler.capacity.node-locality-delay=0
yarn.scheduler.capacity.resource-calculator=org.apache.hadoop.yarn.util.resource.DefaultResourceCalculator
yarn.scheduler.capacity.schedule-asynchronously.maximum-threads=1
yarn.scheduler.capacity.schedule-asynchronously.scheduling-interval-ms=10
yarn.scheduler.capacity.per-node-heartbeat.maximum-offswitch-assignments=4

Tez
---
tez.am.container.reuse.locality.delay-allocation-millis=0;
tez.am.container.reuse.non-local-fallback.enabled=true; //enable for cloud
tez.task.launch.cmd-opts=-XX:+PrintGCDetails -verbose:gc -XX:+UseNUMA -XX:+UseG1GC -XX:+ResizeTLAB
tez.am.launch.cmd-opts=-XX:+PrintGCDetails -verbose:gc -XX:+UseNUMA -XX:+UseG1GC -XX:+ResizeTLAB

HDFS
------
fs.trash.interval=0
fs.trash.checkpoint.interval=0

Testing done:
  - I created custom template from DE & DE HA templates with tuned values in mow-dev. And launched clusters couple of times. There were no issues. Noticed changes were applied.
  - Ran some beeline queries, they worked fine.


Closes #CDPD-6438

sunilgovind commented:
We need to test against YARN nightly.  Please create a cluster with this changes (may be 3/5 nodes) and share us the URL.
And we will request our QE to run YARN nightly against this change. However we may also need to run this change against a canary i think
cc @vinoduec

amitagarwal06 commented:
@sunilgovind 
I have created a 3 node secure cluster one can launch CM UI via mow-dev url (https://cloudera.dps.mow-dev.cloudera.com/cloud/workloads/details/distrox-perf-cluster/events)

And direct url for cluster is https://distrox-perf-cluster-master3.distrox.xcu2-8y8x.dev.cldr.work/distrox-perf-cluster/cdp-proxy/cmf/home/ 

Let me know if you need any other details.

risdenk commented on core/src/main/resources/defaults/blueprints/cdp-data-engineering-702.bp:
```diff
@@ -48,6 +48,10 @@
             "value": "yarn,hive,hdfs"
+          },
+          {
+            "name": "yarn_service_mapred_safety_valve",
```
Should really really avoid safety valves. If this config is important, why isn't it exposed in CM as a real config value?

risdenk commented on core/src/main/resources/defaults/blueprints/cdp-data-engineering-hueha-702.bp:
```diff
@@ -48,6 +48,10 @@
+          },
+          {
+            "name": "yarn_service_mapred_safety_valve",
+            "value": "<property><name>mapreduce.shuffle.connection-keep-alive.enable</name><value>true</value></property>"
```
Should really really avoid safety valves. If this config is important, why isn't it exposed in CM as a real config value?

amitagarwal06 commented:
@sunilgovind I have created a new 3 node secure cluster for you to validate test cases, one can launch CM UI via mow-dev url (https://cloudera.dps.mow-dev.cloudera.com/cloud/workloads/details/distrox-aws-perf-cluster/events)

Direct url for cluster is https://distrox-aws-perf-cluster-master0.cloud-pe.xcu2-8y8x.dev.cldr.work/distrox-aws-perf-cluster/cdp-proxy/cmf/home/

amitagarwal06 commented:
@sunilgovind As discussed test-suites are failing because they depend upon a cluster in mastodon-test or sys-test AWS account. Hence terminated my cluster and provided a custom blueprint name "perf data eng" (with all config changes) to @charanh so that he can launch the cluster in mastodon-test account and run test-suites again.

amitagarwal06 commented:
@sunilgovind Your tests have validated "mapreduce.shuffle.connection-keep-alive.enable=true" but for now I have removed it from the patch as this should be exposed directly as per this JIRA "mapreduce.shuffle.connection-keep-alive.enable=true" not via safety valve.
Also FYI, I have added "yarn.scheduler.capacity.per-node-heartbeat.maximum-offswitch-assignments=4" in this patch but it was already there in your cluster where you ran tests to validate it.

@harishjp Please review Tez changes in this patch.

@mukulhorton Please review fs trash related changes. FYI, This is done only for CDP public cloud as per https://jira.cloudera.com/browse/CDPD-6592

PrabhuJoseph commented:
@Amit Agarwal Thanks for the patch. 

1. The config yarn.scheduler.capacity.default.minimum-user-limit-percent queue-path is wrong. It has to be yarn.scheduler.capacity.root.default.minimum-user-limit-percent if it is only for default queue or yarn.scheduler.capacity.root.minimum-user-limit-percent if it is for all queues under root.

2. The CM template name yarn_scheduler_capacity_resource_calculator is wrong. It has to be 
yarn.scheduler.capacity.resource-calculator

3. yarn.scheduler.capacity.schedule-asynchronously.enable is already set to true in CM for any CDH since 7.0.2. Is this still needed in CB template?

4. Default of yarn.scheduler.capacity.maximum-applications is 10000.

mukul1987 commented:
The changes for skipping the trash look good but it is important to get these documented, this is a change in behavior.

amitagarwal06 commented:
@keyki YARN, Hive & HDFS team has approved it. Pls approve and merge it.

mukul1987 commented:
@steveloughran, can you please have a look at this as well ?

amitagarwal06 commented:
@steveloughran As per the bug https://jira.cloudera.com/browse/CDPD-6592 we are setting
fs.trash.interval=0
fs.trash.checkpoint.interval=0

as default settings for DH clusters. Kindly review.

amitagarwal06 commented:
@keyki All concerned folks have approved it. Pls approve it & merge the change to master so that it is available for Jan 30 release.
